### PR TITLE
Add docker files and travis integration

### PR DESCRIPTION
**Pull request by igchor**
There is no coverity support yet. It will be added in the future.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/4)

### CODE REVIEW

File: utils/docker/images/Dockerfile.fedora-25 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	autoconf \
+	automake \
+	bc \
+	clang \
+	cmake \
+	doxygen \
+	file \
+	findutils \
+	fuse \
+	fuse-devel \
```
Comment (marcinslusarz) [inline utils/docker/images/Dockerfile.fedora-25]: Please figure out the minimal list of packages required for libpmemobj++ and its dependencies. fuse is definitely not a requirement.

File: utils/docker/run-build.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+CC=gcc CXX=g++ \
+cmake .. -DDEVELOPER_MODE=1 \
+			-DCMAKE_BUILD_TYPE=Debug \
+			-DCMAKE_INSTALL_PREFIX=$INSTALL_DIR \
+			-DUSE_LLVM_LIBCPP=0
+
+make -j2
+ctest --output-on-failure
+make install
+
```
Comment (marcinslusarz) [inline utils/docker/run-build.sh]: What about release builds? I think we should try at least one.

**Comment by igchor** [comment]
Review status: 0 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[utils/docker/run-build.sh, line 91 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/4#-LDuy9XXIPFO6BKmpJWa-r1-91:-LE8e-fpDDSiAmN_uEfP:bs7wumb) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/f256ca5aaabf593d524b18e1a244a50ba453a9f3/utils/docker/run-build.sh#L91)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

What about release builds? I think we should try at least one.
</blockquote></details>

Done.

---

*[utils/docker/images/Dockerfile.fedora-25, line 54 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/4#-LDuy9XXIPFO6BKmpJWT-r1-54:-LE8e7bL3FkwrJA4DX-k:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/f256ca5aaabf593d524b18e1a244a50ba453a9f3/utils/docker/images/Dockerfile.fedora-25#L54)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Please figure out the minimal list of packages required for libpmemobj++ and its dependencies. fuse is definitely not a requirement.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/4)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/4#-:-LE8vOqJ-CyAvWK5aSUw:bnfp4nl)*

**Pull request merged**